client/src/main/java/io/pravega/client/stream/impl/StreamSegments.java

```diff
@@ -28,15 +29,19 @@
 public class StreamSegments {
     private static final HashHelper HASHER = HashHelper.seededWith("EventRouter");
     private final NavigableMap<Double, Segment> segments;
+    @Getter
+    private final String delegationToken;
```

Comment:
This is a very odd place for this value. Perhaps it should be moved out into the caller.

Reply:
> The delegation tokens do not contain a timeout, nor any way they could be timed out. IE: There is no API to refresh them or error message for their expiry. If we want this to remain true, we should just make them constants in the reader and writer side that are fetched on the first request for a segment rather than piggybacking them on every single request to the controller.

1. There is an API to refresh the token: `rpc getDelegationToken(StreamInfo) returns (DelegationToken);`
2. The way expiry is handled in the flow is that both timeout expiry as well as any other validation error always terminate the connection. The client is implemented in such a way that any termination of connection results in another call to contoller (as described here: https://github.com/pravega/pravega/wiki/PDP-23:-Pravega-security----encryption-and-Role-Based-Access-Control#c-token-lifetime-and-revocation).
3. The pending thing here is to actually add the timeout while the token is created. Will take this up as another PR.